ts/Series/Map/MapSeries.ts

```diff
@@ -532,6 +532,17 @@ class MapSeries extends ScatterSeries {
         if (this.dataLabelsGroup) {
             this.dataLabelsGroup.clip(this.chart.clipRect);
         }
+
+        this.points.forEach((point): void => {
+            if (point.dataLabel) {
+                if (
+                    point.projectedPath &&
+                    !point.projectedPath.length
+                ) {
+                    point.dataLabel.hide();
+                }
+            }
+        });
```

Comment:
I'm wondering about this code - what if we didn't store `point.projectedPath` in the first place when it is `[]`? Would that stop the data label from being added so that we didn't have to loop over the points and hide conditionally?

Reply:
I have added this condition to hide the data label when the point is not visible in the current range of the map view (sometimes e.g. Brazil data label was shown, but the shape of Brazil wasn't visible due to the current earth's rotation [Orthographic Projection]). Unfortunately, deleting the `point.projectedPath` when it is empty won't stop drawing the data label, but I came up with another idea.

We can use the loop through points inside MapSeries `translate` method, check if `projectedPath` exists and set a proper visible flag. Thanks to that, the data label won't be shown due to the basic logic for drawing data labels and we won't have additional loops.